redis/src/cluster_client.rs

```diff
@@ -177,23 +177,21 @@ impl ClusterClientBuilder {
         } else {
             &None
         };
-        if cluster_params.tls.is_none() {
-            cluster_params.tls = match first_node.addr {
-                ConnectionAddr::TcpTls {
-                    host: _,
-                    port: _,
-                    insecure,
-                    tls_params: _,
-                } => Some(match insecure {
-                    false => TlsMode::Secure,
-                    true => TlsMode::Insecure,
-                }),
-                _ => None,
-            };
-        }
+        let tls = if cluster_params.tls.is_none() {
+            cluster_params.tls = first_node.addr.tls_mode();
+            cluster_params.tls
+        } else {
+            None
+        };
+        let protocol = if cluster_params.protocol.is_none() {
+            cluster_params.protocol = Some(first_node.redis.protocol);
+            cluster_params.protocol
+        } else {
+            None
+        };
 
-        let mut nodes = Vec::with_capacity(initial_nodes.len());
-        for mut node in initial_nodes {
+        // verify that the initial nodes match the cluster client's configuration.
+        for node in initial_nodes.iter() {
```

Comment:
```suggestion
        for node in &initial_nodes {
```

Just a nit -- looks cleaner to me. Ignore if you think otherwise.